tower-retry/src/policy.rs

```diff
@@ -46,7 +44,7 @@ use futures::Future;
 /// ```
 pub trait Policy<Req, Res, E>: Sized {
     /// The `Future` type returned by `Policy::retry()`.
-    type Future: Future<Item = Self, Error = ()>;
+    type Future: Future<Output = Option<Self>>;
```

Comment:
Should we just make this `Output = Self`? We currently handle the case when the future "fails" in `src/future.rs`, but I don't know if that's still relevant? /cc @carllerche

Reply:
Yeah, make this `Self` 👍